tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -1231,6 +1233,101 @@ def get_kurtosis(y):
     return zip(index, res)
 
 
+@set_property("fctype", "simple")
+@set_property("input", "pd.Series")
+@set_property("index_type", pd.DatetimeIndex)
+def energy_content_frequency_brackets(
+    signal: pd.Series,
+    with_normalization: bool = False,
+    bin_index: int = 0,
+    number_of_bins: int = 100,
+    bin_ranges: tuple[int, int] = (-1, -1),
+    return_mode: str = "value",
+) -> float:
+    """_summary_
+
+    Args:
+        signal (pd.Series): _description_
+        with_normalization (bool, optional): _description_. Defaults to False.
+        bin_index (int, optional): _description_. Defaults to 0.
+        number_of_bins (int, optional): _description_. Defaults to 100.
+        bin_ranges (tuple[int, int], optional): _description_. Defaults to (-1, -1).
+        return_mode (str, optional): _description_. Defaults to "value".
+
+    Returns:
+        float: _description_
+    """
+
+    def find_nearest(array: np.ndarray, value: float) -> tuple[int, float]:
+        """Finds the index and value of the value closest to #value in a given
+        #array, and returns both
+
+        Args:
+            array (np.ndarray): Array to be searched in
+            value (float): Target value which the array will be searched for
+
+        Returns:
+            tuple[int, float]: Tuple consisting of index of value and value
+            itself
+        """
+        flt_val: float = float(value)
+        idx: int = int(np.searchsorted(array, flt_val, side="left"))
+        if idx > 0 and (
+            idx == len(array)
+            or math.fabs(flt_val - array[idx - 1]) < math.fabs(flt_val - array[idx])
+        ):
+            return (idx - 1, array[idx - 1])
+        else:
+            return (idx, array[idx])
+
+    time_vec: np.ndarray = signal.index.to_numpy()
+
+    if not np.all(np.isclose(a=np.diff(time_vec), b=np.diff(time_vec)[0])):
+        warnings_message: str = "Uneven time steps detected. This feature can "
+        warnings_message += "only be used for an even distribution of time stamps."
+        warnings.warn(warnings_message, RuntimeWarning)
+        return np.NaN
+
+    number_of_bins: int = 10
+
+    if bin_index >= number_of_bins:
+        bin_index = number_of_bins - 1
+
+    freq_vec: np.ndarray = np.fft.rfftfreq(n=len(signal), d=(time_vec[1] - time_vec[0]))
+    fft_vec: np.ndarray = np.abs(np.fft.rfft(signal.to_numpy()))
+
+    if (bin_ranges[0] < 0 and bin_ranges[1]) < 0 or (bin_ranges[0] >= bin_ranges[1]):
```

Comment:
I guess you want the bracket after the < 0?
```suggestion
    if (bin_ranges[0] < 0 and bin_ranges[1] < 0) or (bin_ranges[0] >= bin_ranges[1]):
```